src/core/layout/qgslayoutexporter.cpp

```diff
@@ -43,26 +57,693 @@ void QgsLayoutExporter::renderPage( QPainter *painter, int page )
   renderRegion( painter, paperRect );
 }
 
-void QgsLayoutExporter::renderRegion( QPainter *painter, const QRectF &region )
+QImage QgsLayoutExporter::renderPageToImage( int page, QSize imageSize, double dpi ) const
+{
+  if ( !mLayout )
+    return QImage();
+
+  if ( mLayout->pageCollection()->pageCount() <= page || page < 0 )
+  {
+    return QImage();
+  }
+
+  QgsLayoutItemPage *pageItem = mLayout->pageCollection()->page( page );
+  if ( !pageItem )
+  {
+    return QImage();
+  }
+
+  QRectF paperRect = QRectF( pageItem->pos().x(), pageItem->pos().y(), pageItem->rect().width(), pageItem->rect().height() );
+  return renderRegionToImage( paperRect, imageSize, dpi );
+}
+
+///@cond PRIVATE
```

Comment:
Can you explain to me the meaning of this notation `///@cond PRIVATE`? Thank you :)

Reply:
It's used to hide code from doxygen. We use it to mark parts of headers (and cpp files) which we don't want doxygen to know about, with the belief that hiding them from doxygen is sufficient enough to dissuade most users from relying on non-stable api.

Reply:
Note that because we have doxygen also scanning cpp files (can't remember why, but there was a good reason for it sometime ago) you need to make sure the cond notation is used both in headers and cpp files.

Reply:
OK, thank you for the explanation, I didn't know that!